libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Solo.cpp

```diff
@@ -78,18 +79,34 @@ void AP_BattMonitor_SMBus_Solo::timer()
     // read the button press indicator
     if (read_block(BATTMONITOR_SMBUS_MANUFACTURE_DATA, buff, 6, false) == 6) {
         bool pressed = (buff[1] >> 3) & 0x01;
-
+        static bool power_off_started = false;
```

Comment:
I don't see why you need this? As far as I can tell monitoring `AP_Notify::flags.powering_off` gets you all the same features without the extra var, and should be more readable.

Reply:
Well, I would have done it that way. But the last time I referenced an AP_Notify flag outside of AP_Notify, everyone told me I should never do that 🙂